swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt

```diff
@@ -280,7 +280,7 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
       val maxItemsToProcess = Math.min(preProcessedCandidates.size, getMaxItemsProcessedPerCycle(workConfiguration))
 
       for (candidate in preProcessedCandidates) {
-        if (candidateCounter.get() > maxItemsToProcess) {
+        if (candidateCounter.get() >= maxItemsToProcess) {
```

Comment:
Makes sure to exit the loop when the maxItemsToProcess is reached.

Reply:
HA 😄